### ISSUE TITLE
Doesn't work with Vite 3

### ISSUE THREAD

**Issue by brillout** [preactjs/preact#3558]
- [x] Check if updating to the latest Preact version resolves the issue

**Describe the bug**

Preact doesn't work with Vite 3 (`3.0.0-alpha.9`).

**To Reproduce**

See https://github.com/brillout/preact-vite3.

CodeSandbox: https://codesandbox.io/s/github/brillout/preact-vite3

**Expected behavior**

Preact seems to inject a `require` statement which it shouldn't because Vite 3 is now ESM by default.

**Comment by JoviDeCroock** [comment]
This might be because of using the wrong entry in the export map, we only use import/export in our esm-bundles.

**Comment by brillout** [comment]
Thanks @JoviDeCroock for the answer.

The vite-plugin-ssr test suite tests a lot of integration, and only `urql` and `preact` are failing. That's why I'm leaning towards thinking it's a problem on `preact`'s side.

The error, and `grep require` in `dist/server/`, are quite insightful.

**Comment by JoviDeCroock** [comment]
Yes, so I did run that and it looks like it's a transitive import issue when the jsx import is included so this issue could be located in the `@preact/preset-vite` repository

**Comment by brillout** [comment]
Yes exactly that's the problematic `require` statement. I also tried to figure out why it's there but no clue 😅.

**Comment by JoviDeCroock** [comment]
As far as I can tell this is an issue with `cjs-module-lexer` and `export *` is missinterpreted by it, it does not see them as `reexports`. This was seen in urql as well https://github.com/FormidableLabs/urql/pull/2485

**Comment by JoviDeCroock** [comment]
Apologies that was accidental while mentioning this issue in the PR

**Comment by JoviDeCroock** [comment]
This might solve your issue as well @brillout https://github.com/preactjs/preset-vite/pull/50
